src/library/scala/collection/immutable/ListMap.scala

```diff
@@ -79,6 +80,17 @@ sealed class ListMap[K, +V]
     res
   }
 
+  /** Apply `f` to each key/value pair for its side effects
+   * Note: [U] parameter needed to help scalac's type inference.
+   */
+  override def foreachEntry[U](f: (K, V) => U): Unit = {
+    var curr: ListMap[K, V] = this
+    while (curr.nonEmpty) {
+      f(curr.key, curr.value)
+      curr = curr.next
+    }
+  }
```

Comment:
Isn't this going to present key-values in the wrong order, since ListMap is stored in reverse?

Reply:
You are right. This is okay for the `hashCode` computation (which is order-agnostic), but doesn't seem suitable for the general `foreachEntry` method.

Reply:
I've now removed this override other than in a private helper class used in `ListMap.hashCode`. I've also overrided `foreachEntry` in `SeqMapN`.